src/test/java/com/google/sps/servlets/LiveCourseDataTest.java

```diff
@@ -0,0 +1,142 @@
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+public final class LiveCourseDataTest {
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private AddSchoolData newSchoolObject;
+  private LiveCourseData LiveData;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    newSchoolObject = new AddSchoolData();
+    LiveData = new LiveCourseData();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Mock HttpServletRequest request;
+  @Mock HttpServletRequest requestB;
+
+  @Test
+  public void FindingExisitngTermEntity() {
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    request =
+        createRequest(
+            request, /*schoolName*/
+            "MIT", /*courseName*/
+            "6.006", /*termName*/
+            "Spring 2020", /*units*/
+            "12", /*profName*/
+            "Jason Ku");
+    requestB = createRequest(requestB, "MIT", "6.008", "Spring 2018", "6", "Srini");
+    String expectedTermName = "Spring 2020";
+
+    newSchoolObject.addSchoolData(db, request);
+    newSchoolObject.addSchoolData(db, requestB);
+    Key expectedParent = findQueryMatch(db, "Course", "course-name", "6.006").get(0).getKey();
+    Entity found = LiveData.getTerm(db, request);
+
+    assertEquals(expectedParent, found.getParent());
+    assertEquals(expectedTermName, found.getProperty("term"));
+  }
+
+  @Test
+  public void GettingRatingData_Hours_Difficulty() {
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    Entity parent = addTermEntity(db);
+    addRatingEntity(db, 12, 7, parent.getKey());
+    addRatingEntity(db, 7, 4, parent.getKey());
+
+    List<Long> expectedHoursList = new ArrayList(Arrays.asList((long) 12, (long) 7));
+    List<Long> expectedDifficultyList = new ArrayList(Arrays.asList((long) 7, (long) 4));
+
+    List<Long> actualHoursList = LiveData.getTermData(db, parent, "hours");
+    List<Long> actualDifficultyList = LiveData.getTermData(db, parent, "difficulty");
+
+    assertEquals(expectedHoursList, actualHoursList);
+    assertEquals(expectedDifficultyList, actualDifficultyList);
+  }
+
+  private HttpServletRequest createRequest(
+      HttpServletRequest request,
+      String schoolName,
+      String courseName,
+      String termName,
+      String units,
+      String profName) {
+    request = Mockito.mock(HttpServletRequest.class);
```

Comment:
If you're already declaring a new on here, why pass it in as a parameter? I don't think that's necessary. 

Also both returning a type while also asking for the same type in the params confuses people. It's not clear then if mutation is going on or what exactly.

Reply:
Nice catch, I was even confused as to why I did that.

Reply:
I actually did a little more testing. And what I am doing now is passing in the @Mock request and @Mock requestB I define at the top. I then reassign these inside of the function. When I try doing it the simpler way of just assigning request = createRequest(...) and requestB = createRequest(...) I get an IndexOutOfBounds error. Not sure why this is the case.

Reply:
Huh, weird. Well as long as you're consistent it's fine.

Reply:
I see you've requested a re-review but I don't see the changes mentioned here yet?

Reply:
Wait is there an error here outside of me passing in the request?

Reply:
My understanding from your comment is that you were going to change this to just mutate the request (which is what you are already doing).

However that implies you'd change where this is used to not expect a return object (and also change the return type of this function to void). 

I'll comment where it's use to further explain.